inst/@sym/kron.m

```diff
@@ -0,0 +1,102 @@
+%% Copyright (C) 2016 Utkarsh Gautam
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @deftypefn  {Function File} {@var{c} =} kron (@var{a}, @var{b})
+%% Kronecker tensor product of two matrices.
+%%
+%% Examples:
+%% @example
+%% @group
+%% >> syms x
+%% >> kron(eye(2)*x,[1,-1;-1,1])
+%% @result{} ans = (sym 4×4 matrix)
+%% 	
+%%     ⎡x   -x  0   0 ⎤
+%%     ⎢              ⎥
+%%     ⎢-x  x   0   0 ⎥
+%%     ⎢              ⎥
+%%     ⎢0   0   x   -x⎥
+%%     ⎢              ⎥
+%%     ⎣0   0   -x  x ⎦
+%%			
+%%
+%% @end group
+%% @end example
+%
+%% @example
+%% @group
+%% >> syms x y
+%% >> kron([1,2],[x,y;y,x])
+%% @result{} ans = (sym 2×4 matrix)
+%%
+%%     ⎡x  y  2⋅x  2⋅y⎤
+%%     ⎢              ⎥
+%%     ⎣y  x  2⋅y  2⋅x⎦
+%%						
+%% @end group
+%% @end example
+%% @end deftypefn
+%% Author: Utkarsh Gautam
+%% Keywords:  kron product
+function c = kron(a, b)
+
+	if ~(ismatrix(a) && ismatrix(b))
+		error('Invalid Input');
+	end	
+  cmd = { 'a, b = _ins'
+  		    'from sympy.physics.quantum import TensorProduct'
+          'return TensorProduct(Matrix(a),Matrix(b)),'
```

Comment:
does this work if either `x` or `y` is a scalar?  It needs tests for this case.  I suspect it does not because of the `Matrix(a)` bit.  Maybe need something like  `if not a.is_Matrix: a = Matrix([a])`